### PR TITLE
MRG raw.filter copy param

### PR DESCRIPTION
**Pull request by jona-sassenhagen**
A start. Partially adresses #3070.
As simple as this right? @teonbrooks @kingjr 

~~Missing test.~~ (Actually simplifies the tests ...)

### CODE REVIEW

File: mne/io/base.py [hunk 1]
```diff
@@ -778,6 +778,9 @@ def filter(self, l_freq, h_freq, picks=None, filter_length='10s',
             Dictionary of parameters to use for IIR filtering.
             See mne.filter.construct_iir_filter for details. If iir_params
             is None and method="iir", 4th order Butterworth will be used.
+        copy : bool
+            If `True`, modify and return a copy. If `False`, modify in-place
```
Comment (kingjr) [inline mne/io/base.py]: 'Returns a modified copy' seems less ambiguous to me

**Comment by kingjr** [comment]
Updates what's new for api changes. Otherwise lgtm!

File: mne/io/fiff/tests/test_raw_fiff.py [hunk 2]
```diff
@@ -717,17 +717,11 @@ def test_filter():
...
-    raw_bp = raw.copy()
-    raw_bp.filter(4.0 + 0.25, 8.0 - 0.25, picks=picks)
-
-    raw_bs = raw.copy()
-    raw_bs.filter(8.0 + 0.25, 4.0 - 0.25, picks=picks, n_jobs=2)
+    filter_params = dict(picks=picks, n_jobs=2, copy=True)
+    raw_lp = raw.filter(0., 4.0 - 0.25, **filter_params)
+    raw_hp = raw.filter(8.0 + 0.25, None, **filter_params)
+    raw_bp = raw.filter(4.0 + 0.25, 8.0 - 0.25, **filter_params)
+    raw_bs = raw.filter(8.0 + 0.25, 4.0 - 0.25, **filter_params)
```
Comment (agramfort) [inline mne/io/fiff/tests/test_raw_fiff.py]: are you actually testing the copy=False option?

Comment (jona-sassenhagen) [inline mne/io/fiff/tests/test_raw_fiff.py]: Yes, in line 757.

**Comment by agramfort** [comment]
ok thx

travis is not happy

**Comment by jona-sassenhagen** [comment]
Please merge before anybody touches whats_new and I have to rebase ;)

File: mne/io/base.py [hunk 3]
```diff
@@ -778,6 +778,9 @@ def filter(self, l_freq, h_freq, picks=None, filter_length='10s',
             Dictionary of parameters to use for IIR filtering.
             See mne.filter.construct_iir_filter for details. If iir_params
             is None and method="iir", 4th order Butterworth will be used.
+        copy : bool
+            If ``True``, return a modified copy. If ``False``, modify
+            in-place and return the original object.
```
Comment (Eric89GXL) [inline mne/io/base.py]: the modified original object

File: doc/whats_new.rst [hunk 4]
```diff
@@ -95,7 +95,7 @@ BUG
 
     - Fix bug with :func:`mne.label_sign_flip` (and :func:`mne.extract_label_time_course`) by `Natalie Klein`_ and `Eric Larson`_
 
-    - Add copy parameter in :func:`mne.Epochs.apply_baseline` method by `Jona Sassenhagen`_ and `Alex Gramfort`_
+    - Add copy parameter in :func:`mne.Epochs.apply_baseline` and :func:`mne.raw.filter` methods by `Jona Sassenhagen`_ and `Alex Gramfort`_
```
Comment (Eric89GXL) [inline doc/whats_new.rst]: `mne.io.Raw.filter`

**Comment by jona-sassenhagen** [comment]
Done @Eric89GXL

File: mne/io/base.py [hunk 5]
```diff
@@ -779,8 +779,8 @@ def filter(self, l_freq, h_freq, picks=None, filter_length='10s',
             See mne.filter.construct_iir_filter for details. If iir_params
             is None and method="iir", 4th order Butterworth will be used.
         copy : bool
-            If ``True``, return a modified copy. If ``False``, modify
-            in-place and return the original object.
+            If ``True``, return the modified original object. If ``False``,
```
Comment (Eric89GXL) [inline mne/io/base.py]: I think `return a modified copy` was less ambiguous, but I like the second part better now. It seems like a minor detail but we don't want people to get hung up on this.

Comment (Eric89GXL) [inline mne/io/base.py]: I'll fix in `master`

**Comment by Eric89GXL** [comment]
Thanks @jona-sassenhagen

**Comment by jona-sassenhagen** [comment]
Thanks Eric Alex Jean-Remi

**Pull request merged**